server/bed.js

```diff
@@ -47,23 +47,28 @@ bed.MotifsResult;
 /** @const */
 bed.query = {};
 
+// Start public APIs
 /**
- * @typedef {{
+ * @param {*|{
  *   fileName: string,
  *   chr: string,
  *   xl: (number|undefined),
  *   xr: (number|undefined)
- * }}
- */
-bed.query.Motifs;
-
-// Start public APIs
-/**
- * @param {!bed.query.Motifs} query
+ * }} query
  * @param {string} bedPath
  * @return {bed.MotifsResult|bed.Error}
  */
 bed.query.motifs = function(query, bedPath) {
+  if (query.fileName == undefined) {
```

Comment:
Use === (triple-equal) to compare with undefined, so as to differ from comparing with null. Sometimes null has different meanings.